server/service/src/sensor/berlinger.rs

```diff
@@ -0,0 +1,380 @@
+    sensor_id: &str,
+) -> Result<Vec<TemperatureBreach>, RepositoryError> {
+    TemperatureBreachRepository::new(connection).query_by_filter(
+        TemperatureBreachFilter::new()
+            .sensor(SensorFilter::new().id(EqualFilter::equal_to(sensor_id)))
+            .acknowledged(false),
+    )
+}
+
+pub fn get_breach_configs_for_sensor(
+    connection: &StorageConnection,
+    _sensor_id: &str,
+) -> Result<Vec<TemperatureBreachConfig>, RepositoryError> {
+    TemperatureBreachConfigRepository::new(connection)
+        .query_by_filter(TemperatureBreachConfigFilter::new().is_active(true))
+}
+
+fn sensor_add_log_if_new(
+    connection: &StorageConnection,
+    sensor_row: &SensorRow,
+    temperature_log: &temperature_sensor::TemperatureLog,
+) -> Result<(), RepositoryError> {
+    let result = get_matching_sensor_log(connection, &sensor_row.id, temperature_log.timestamp)?;
+
+    if let Some(_record) = result.clone().pop() {
+        Ok(())
+    } else {
+        let new_temperature_log = TemperatureLogRow {
+            id: uuid(),
+            store_id: sensor_row.store_id.clone(),
+            sensor_id: sensor_row.id.clone(),
+            location_id: sensor_row.location_id.clone(),
+            temperature: temperature_log.temperature,
+            datetime: temperature_log.timestamp,
+            temperature_breach_id: None,
+        };
+        TemperatureLogRowRepository::new(connection).upsert_one(&new_temperature_log)?;
+        println!("Added sensor log {:?} ", new_temperature_log);
+        Ok(())
+    }
+}
+
+fn sensor_add_breach_if_new(
+    connection: &StorageConnection,
+    sensor_row: &SensorRow,
+    temperature_breach: &temperature_sensor::TemperatureBreach,
+    breach_config: &temperature_sensor::TemperatureBreachConfig,
+) -> Result<(), RepositoryError> {
+    let breach_row_type = get_breach_row_type(&temperature_breach.breach_type);
+    let result = get_matching_sensor_breach(
+        connection,
+        &sensor_row.id,
+        temperature_breach.start_timestamp,
+        temperature_breach.end_timestamp,
+        &breach_row_type,
+    )?;
+
+    if let Some(mut record) = result.clone().pop() {
+        if record.temperature_breach_row.end_datetime != Some(temperature_breach.end_timestamp) { // Update breach end time if it has changed
+            record.temperature_breach_row.end_datetime = Some(temperature_breach.end_timestamp);
+            TemperatureBreachRowRepository::new(connection).upsert_one(&record.temperature_breach_row)?;
+        }
+        Ok(())
+    } else {
+        let new_temperature_breach = TemperatureBreachRow {
+            id: uuid(),
+            store_id: sensor_row.store_id.clone(),
+            sensor_id: sensor_row.id.clone(),
+            location_id: sensor_row.location_id.clone(),
+            start_datetime: temperature_breach.start_timestamp,
+            end_datetime: Some(temperature_breach.end_timestamp),
+            acknowledged: false,
```

Comment:
Does berlinger store breach acknowledgement status ? You can acknowledge breaches on the device right ?

Reply:
As far as I know you can (https://smartview.zendesk.com/hc/en-us/articles/360018899919-How-do-I-confirm-an-Alarm-on-my-Fridge-tag) although I haven't actually tried it

Reply:
Does it come through in the .txt data ?